src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -416,6 +423,41 @@ def _remove_grpc_attributes(cls, values: Dict[str, Any]) -> Dict[str, Any]:
 
         return values
 
+    @root_validator
+    def _validate_backup_strategy(
+        cls, values: Dict[str, Any]
+    ) -> Dict[str, Any]:
+        """Validate the backup strategy.
+
+        Args:
+            values: All model attribute values.
+
+        Returns:
+            The model attribute values.
+
+        Raises:
+            ValueError: If the backup database name is not set when the backup
+                database is requested.
+        """
+        backup_strategy = values.get("backup_strategy")
+        if backup_strategy == DatabaseBackupStrategy.DATABASE and (
+            not values.get("backup_database")
+        ):
+            raise ValueError(
+                "The backup database name must be set if the backup strategy "
+                "is set to use a backup database."
+            )
+
+        if backup_strategy == DatabaseBackupStrategy.DUMP_FILE and (
+            not values.get("backup_directory")
+        ):
+            values["backup_directory"] = os.path.join(
+                GlobalConfiguration().config_directory,
+                SQL_STORE_BACKUP_DIRECTORY_NAME,
```

Comment:
should this also maybe validate that there are sufficient permissions to write to this directory?

Reply:
we validate it by writing to it